token/js/src/actions/transfer.ts

```diff
@@ -28,13 +31,27 @@ export async function transfer(
     amount: number | bigint,
     multiSigners: Signer[] = [],
     confirmOptions?: ConfirmOptions,
-    programId = TOKEN_PROGRAM_ID
+    programId = TOKEN_PROGRAM_ID,
+    memoProgramId = MEMO_PROGRAM_ID,
+    memo?: string
 ): Promise<TransactionSignature> {
     const [ownerPublicKey, signers] = getSigners(owner, multiSigners);
 
     const transaction = new Transaction().add(
         createTransferInstruction(source, destination, ownerPublicKey, amount, multiSigners, programId)
     );
 
+    // Add an (optional) note describing the transaction.
+    // Don't bother adding if memo is an empty string
+    if (memo?.length && memoProgramId) {
+        await transaction.add(
+            new TransactionInstruction({
+                keys: [{ pubkey: source, isSigner: true, isWritable: true }],
+                data: Buffer.from(memo, 'utf-8'),
+                programId: new PublicKey(memoProgramId),
+            })
```

Comment:
Rather than redefining this here, how about using the instruction builder and constant from `@solana/spl-memo` https://github.com/solana-labs/solana-program-library/blob/master/memo/js/src/index.ts ?